src/ProxyInterfaceSourceGenerator/SyntaxReceiver/ProxySyntaxReceiver.cs

```diff
@@ -6,26 +6,40 @@
 
 namespace ProxyInterfaceSourceGenerator.SyntaxReceiver;
 
-internal class ProxySyntaxReceiver : ISyntaxReceiver
+internal class ProxySyntaxReceiver : ISyntaxContextReceiver
 {
-    private static readonly string[] Modifiers = { "public", "partial" };
     private static readonly string[] GenerateProxyAttributes = { "ProxyInterfaceGenerator.Proxy", "Proxy" };
-
+    private static readonly string[] Modifiers = { "public", "partial" };
     public IDictionary<InterfaceDeclarationSyntax, ProxyData> CandidateInterfaces { get; } = new Dictionary<InterfaceDeclarationSyntax, ProxyData>();
 
-    public void OnVisitSyntaxNode(SyntaxNode syntaxNode)
+    public void OnVisitSyntaxNode(GeneratorSyntaxContext context)
     {
-        if (syntaxNode is InterfaceDeclarationSyntax interfaceDeclarationSyntax && TryGet(interfaceDeclarationSyntax, out var data))
+        var syntaxNode = context.Node;
+        var semanticModel = context.SemanticModel;
+
+        if (syntaxNode is InterfaceDeclarationSyntax interfaceDeclarationSyntax && TryGet(interfaceDeclarationSyntax, out var data, semanticModel!))
         {
             CandidateInterfaces.Add(interfaceDeclarationSyntax, data);
         }
     }
 
-    private static bool TryGet(InterfaceDeclarationSyntax interfaceDeclarationSyntax, [NotNullWhen(true)] out ProxyData? data)
+    private static string ConvertTypeName(string typeName)
+    {
+        return !(typeName.Contains('<') && typeName.Contains('>')) ?

```

Comment:
This method seems to be broken for classes with more than one generic parameter.
`public class Generic<T1, T1>`
I found some code and the GetTypeByMetadataName works now.
Sadly this is now a breaking chance, because the filenames change.

Reply:
Thanks for the fix.

It's not a big issue that the filenames change, these filenames are used internally.